### ISSUE TITLE
US NY Counties - 17 counties reported less infections on 04-23-2020.csv than previous day

### ISSUE THREAD

**Issue by rahmatiangit** [CSSEGISandData/COVID-19#2357]
FYI - so data for New York State has not changed much from yesterday

----------------------------------------------------------
NY_county | less/more infection on 4/23 than yesterday 4/22 

-- | --
Albany | "+"
Allegany | "="
Broome | "+"
Cattaraugus | "="
Cayuga | "+"
**Chautauqua | "-"**
Chemung | "+"
Chenango | "+"
**Clinton | "-"**
Columbia | "+"
Cortland | "="
**Delaware | "-"**
Dutchess | "+"
Erie | "+"
Essex | "="
Franklin | "="
Fulton | "="
Genesee | "+"
Greene | "+"
Hamilton | "="
Herkimer | "+"
Jefferson | "+"
Lewis | "="
Livingston | "+"
**Madison | "-"**
Monroe | "+"
Montgomery | "+"
Nassau | "+"
**New York | "-"**
Niagara | "+"
Oneida | "+"
**Onondaga | "-"**
Ontario | "="
**Orange | "-"**
Orleans | "+"
**Oswego | "-"**
Otsego | "+"
**Putnam | "-"**
**Rensselaer | "-"**
Rockland | "+"
**Saratoga | "-"**
**Schenectady | "-"**
Schoharie | "+"
**Schuyler | "-"**
Seneca | "="
Saint Lawrence | "+"
**Steuben | "-"**
Suffolk | "+"
**Sullivan | "-"**
Tioga | "+"
**Tompkins | "-"**
**Ulster | "-"**
Warren | "+"
Washington | "+"
Wayne | "+"
Westchester | "+"
**Wyoming | "-"**
Yates | "="

Big thanks for collecting all the data (even with some errors here and there)

**Comment by hayesb-emmanuel** [comment]
Yea, total confirmed reported for NY state in the last file on 04/22 was 263,513 and on 04/23 it was 263,460. Seeing on 04/22 there were around 5000 confirmed cases, and there have been between 4000 and 5000 reported cases in the previous 3 days, 04/23 showing a decrease of -53 cases means there was likely incorrectly entered data for NY on 04/23.

**Comment by degelsun** [comment]
For Onondage county, there was a decrease of 124, which can greatly bias statistical results.

**Comment by hayesb-emmanuel** [comment]
That doesn't appear to make up the difference, the total confirmed cases for the state of NY look like this (new confirmed cases per day) ...

04/19: 5847
04/20: 4774
04/21: 5293
04/22: 5029
04/23: -53

It is statistically unlikely that NY was having a daily increase of nearly 4000-5000 per day and then dropping to -53 cases on the 23rd. Confirmed cases will never go down, its just a growing statistic, unless there are errors, but 5082 count of errors wipes out an entire days growth.

I checked and the New York City entry was 1442 less on the 23rd than on the 22nd. A total of 17 lines for New York were lower on the 23rd than reported on the 22nd.

**Comment by hayesb-emmanuel** [comment]
May be related to this ...

https://github.com/CSSEGISandData/COVID-19/issues/2361

Someone else noticed the 4/23 file has timestamps on updates from 4/22, so it could be older data from earlier in the day on the 22nd, like a rolling back of the values.

**Comment by degelsun** [comment]
No. Onondaga county had never have 600 this number of confirmed cases. It had 598 in April 17 and 624 in April 18.

**Comment by hayesb-emmanuel** [comment]
Point I am making is that there is a larger data point issue for the entire state including Onondaga. It's one thing if a county reports a correction, but 5000+ is a bit large for an error rate as it wipes out an entire days values.

**Comment by degelsun** [comment]
It says that data source includes 1Point3Arces (https://coronavirus.1point3acres.com/en). However, data on 1Point3Arces is correct. And this mistake is only presented in NY. 

Onondaga county only updates data once a day (around 4pm). So, 600 is a impossible number. So, I wonder where those number came from.

**Comment by hayesb-emmanuel** [comment]
Does the https://coronavirus.1point3acres.com/en site let you see what the data was on 04/23 as provided by their data feeds? Might look right now, but could have been wrong when the automation pulled the data, then was corrected. Timing makes a difference.

If the data today is valid, and its just the data on 04/23 that is wrong in this repository then they likely won't fix it, and I will just have to find the first accurate data point after the correction to use for 04/23.

**Comment by hayesb-emmanuel** [comment]
Ultimately the NY numbers in the github history for the 22nd vs the 23rd are off by nearly 4000-5000 cases for the state based on the last historical snapshot for each day (24 hour period). I just want to make sure that today's counts are not reflecting a larger issue.

598 vs 600 vs 624 your talking about less than a 5% error rate on the data in a 24 hour period, not likely going to affect analysis by a major degree.

**Comment by jjbenes** [comment]
Based on what I observed in the JHU and the USAFacts COVID-19 data, it must be difficult to get 100% monotonically increasing data for cumulative cases across 3100+ counties in the US alone.

Volunteers can help clean up historical data. I ran the checking code in [cell 4 here](https://nbviewer.jupyter.org/github/jjbenes/covid19/blob/master/jupyter/covid19_jhu_csse.ipynb#Quick-Sanity-Checks) and found 31.9% of the county cumulative cases drop from one day to the next in the time series from 22nd of Jan. to 25th of Apr. (Between 22nd of Jan. to 18th of Apr, 25.4% of the "cumulative" case counts dipped from one day to the next.)

JHU CSSE Monotonicity Check (26th Apr, 2020)
![Screen Shot 2020-04-26 at 2 30 48 PM](https://user-images.githubusercontent.com/62985029/80320395-03cb7e00-87cb-11ea-80f0-8494b6e68e29.png)

I uploaded an [API](https://github.com/jjbenes/covid19) for people to do this sort of checks and plot graphs in a few lines of code. It also works with data from USAFacts.org. Their data come from county health departments listed on their website. They partitioned their data strictly by counties (as opposed to, say, lumping data from all five boroughs of NYC to Manhattan). They also seem to have population numbers that match those from the U.S. Census Bureau. I can't quite say that for the JHU data.

USAFacts.org Monotonicity Check (26th Apr, 2020)
![Screen Shot 2020-04-26 at 2 48 53 PM](https://user-images.githubusercontent.com/62985029/80320801-69206e80-87cd-11ea-9b3e-ac0c15bc4e19.png)

Clearly, pointing out out problems is much easier than fixing them. I don't have the know-how to clean up the data for modeling purposes but I hope that there are volunteers here who can. I can at least provide the tools for them to analyze data from multiple sources. Let me know if you want to work on this.
